src/core/bst.h

```diff
@@ -165,7 +165,55 @@ class BST_stmt_with_dest : public BST_stmt {
     bool has_dest_vreg() const override { return true; }
 };
 
+#define BSTVARVREGS(opcode, base_class, num_elts, vreg_dst)                                                            \
```

Comment:
@kmod Is this what you had in mind? 
I also tried to reduce the code duplication using macros... But in general I don't worry too much about the Nodes currently because I'm currently working on a python script which auto generates them in order to support the 1 and 4 byte vregs...